server/testserver.go

```diff
@@ -63,6 +63,22 @@ func StartTestServer(t util.Tester) *TestServer {
 	return s
 }
 
+// StartTestServerJoining starts an in-memory test server that attempts to join `other`.
+func StartTestServerJoining(t util.Tester, other *TestServer) *TestServer {
```

Comment:
this doesn't use any unexported fields, is there any reason it needs to be linked into the production binary?